zdm/iteration.py

```diff
@@ -194,19 +215,41 @@ def update_grid(grid,pset):
 	alpha=pset[2]
 	gamma=pset[3]
 	sfrn=pset[4]
-	
+	H0value=pset[8]
+
 	# assumes all parameters following this are passed to DM distribution
 	smear_mean=pset[5]
 	smear_sigma=pset[6]
-	oldlEmin=np.log10(grid.Emin)
-	oldlEmax=np.log10(grid.Emax)
+
+	#taking all the old grid parameters
+    oldmask=grid.smear
+    oldsmeargrid=grid.smear_grid
+    
+	oldlEmin=np.log10(float(grid.Emin))
+	oldlEmax=np.log10(float(grid.Emax))
 	oldalpha=grid.alpha
 	oldgamma=grid.gamma
 	oldsfrn=grid.sfr_n
 	oldsmean=grid.smear_mean
 	oldssigma=grid.smear_sigma
+	oldH0value=grid.H0
 	new_sfr_smear=False #do not need to calculate new product of sfr and smearing
 	
+	##check if H0 value changed
+    
+    if oldH0value != pset[8]:
```

Comment:
maybe use `H0value` instead of `pset[8]` for easier comprehension

Reply:
Yep, should've done that

Reply:
OK, so H0 is now added at the end of the parameter set? Can you double check there are no lines with pset[-1] anywhere in the code? I might have used this somewhere to refer to the constant.

Reply:
Ah okay, I will do that! Will make the necessary changes. Thanks!